pkg/asset/installconfig/aws/subnet.go

```diff
@@ -0,0 +1,165 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var vpc string
+var vpcFromSubnet string
+var routeTables []*ec2.RouteTable
+var subnetMetadata map[string]*Subnet
+
+// Subnet hold metadata for a subnet.
+type Subnet struct {
+	// ARN is the subnet's Amazon Resource Name.
+	ARN string
+
+	// Public is true if the cloud-provider logic considers the subnet
+	// to be public (e.g. with suitable routing for hosting public
+	// load balancers).
+	Public bool
+
+	// VPC is the ID of the VPC to which the subnet belongs.
+	VPC string
+
+	// Zone is the subnet's availability zone.
+	Zone string
+}
+
+// SubnetMetadata retrieves metadata for the given subnet(s).  Results
+// are cached, so additional requests for a given subnet do not result
+// in further AWS API calls.
+func SubnetMetadata(ctx context.Context, ids []string) (map[string]Subnet, error) {
```

Comment:
Almost all consumers are having to iterate to split the public from private, could we returns 2 lists??

also we can keep the uniquefication internal to the function and return lists maybe.

Reply:
> Almost all consumers are having to iterate to split the public from private, could we returns 2 lists??

Done since the reroll to build on #2512.

> also we can keep the uniquefication internal to the function and return lists maybe.

It's still returning maps, which allows me to keep the IDs out of the `Subnet` metadata type.  Is there a benefit to using slices instead of maps?